main.cpp

```diff
@@ -18,16 +18,23 @@ using namespace clang::ast_matchers;
 using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
+private:
+    Rewriter& rewriter_;
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+    CastCallBack(Rewriter& rewriter) : rewriter_(rewriter) {}
+
+    void run(const MatchFinder::MatchResult& Result) override {
+        const auto* styleCastExpr = Result.Nodes.getNodeAs<CStyleCastExpr>("cast");
+        const auto locationOpenBracket = styleCastExpr->getLParenLoc();
+        const auto locationCloseBracket = styleCastExpr->getRParenLoc();
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
+        rewriter_.ReplaceText(locationOpenBracket, 1, "static_cast<");
+        rewriter_.ReplaceText(locationCloseBracket, 1, ">(");
+        rewriter_.InsertTextAfterToken(locationCloseBracket.getLocWithOffset(1), ")");

```

Comment:
This code does not work as expected for the following example:
```cpp
// test.cpp
int main() {
    float fooooo;
    int i = (int)   fooooo   ;
    return 0;
}
```

Reply:
This can be fixed by adding a function **_Lexer::getLocForEndOfToken()_** like this:

auto endOfLine = Lexer::getLocForEndOfToken(styleCastExpr->getEndLoc(), 0, *Result.SourceManager, 
Result.Context->getLangOpts());

rewriter.InsertText(endOfLine, ")");